test/list/reduce.wren

```diff
@@ -0,0 +1,9 @@
+var a = [1, 4, 2, 1, 5]
+var max = new Fn {|a, b| a > b ? a : b }
+var sum = new Fn {|a, b| a + b }
```

Comment:
No, foldr is useful, it just does something a little different. The trick is to view lists as trees rather than as lists. One example of an function that would get different results would be exponentiation. Here's the diagrams that I always have to look at:

https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29#Folds_as_structural_transformations